concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Common.hs

```diff
@@ -77,3 +135,34 @@ sigKeyPair = fst $ Dummy.randomBlockKeyPair $ mkStdGen 42
 -- | The public key of the 'sigKeyPair'.
 sigPublicKey :: Sig.VerifyKey
 sigPublicKey = Sig.verifyKey sigKeyPair
+
+-- | Run tests for each protocol version.
+-- The tests are provided with the protocol version and a string representation of the protocol version.
+forEveryProtocolVersion ::
+    (forall pv. (IsProtocolVersion pv) => SProtocolVersion pv -> String -> Spec) ->
+    Spec
+forEveryProtocolVersion check =
+    sequence_
+        [ check SP1 "P1",
```

Comment:
Why don't we just use the `Show` instance for the `ProtocolVersion` ?

Reply:
If found this simpler, it becomes something like this then:
```suggestion
        [ check SP1 (show $ demoteProtocolVersion SP1),
```
I don't mind changing it, if you prefer this.